src/robot/libdocpkg/htmlwriter.py

```diff
@@ -15,11 +15,16 @@
 
 import json
 
+from robot.errors import DataError
 from robot.htmldata import HtmlFileWriter, ModelWriter, LIBDOC
 
 
 class LibdocHtmlWriter(object):
 
+    def __init__(self, spec_doc_format):
+        if spec_doc_format != 'HTML':
+            raise DataError("Spec doc format must be 'HTML' when using 'HTML' output, got '%s'." % spec_doc_format)
```

Comment:
In my opinion --specdocformat should be invalid with HTML format regardless the value. That could be easily validated when handling arguments in the main Libdoc program.

Reply:
i think that the validation of a format accept that or not can be should be done in the writer itself.
and i definitely not think that `libdoc -f HTML -s HTML` should be invalid!
Because sometimes you want to script that and imho there is not need for a failure here.

So if you would make a batch or shellscript for that you would have to get rid of that `-s` if the other value is HTML wich just make live harder and do not add any value.

imho we could just not fail anyway and always just create HTMl, but you wanted that error.